internal/v1/server.go

```diff
@@ -109,21 +110,27 @@ func Attach(conf *ServerConfig) error {
 	s.echo.Binder = binder{}
 	s.echo.HTTPErrorHandler = s.HTTPErrorHandler
 
-	middlewares := []echo.MiddlewareFunc{
+	middlewaresNoAuth := []echo.MiddlewareFunc{
 		prometheus.StatusMiddleware,
 	}
 
+	var middlewares []echo.MiddlewareFunc
+
 	if s.fedoraAuth {
-		middlewares = append(middlewares, echo.WrapMiddleware(fedora_identity.Extractor))
+		middlewares = append(middlewaresNoAuth, echo.WrapMiddleware(fedora_identity.Extractor))
 	} else {
-		middlewares = append(middlewares, echo.WrapMiddleware(identity.Extractor), echo.WrapMiddleware(identity.BasePolicy))
-
+		middlewares = append(middlewaresNoAuth, echo.WrapMiddleware(identity.Extractor), echo.WrapMiddleware(identity.BasePolicy))
 	}
+
+	middlewaresNoAuth = append(middlewaresNoAuth, prometheus.PrometheusMW)
 	middlewares = append(middlewares, s.noAssociateAccounts, s.ValidateRequest, prometheus.PrometheusMW)
 
 	RegisterHandlers(s.echo.Group(fmt.Sprintf("%s/v%s", RoutePrefix(), majorVersion), middlewares...), &h)
 	RegisterHandlers(s.echo.Group(fmt.Sprintf("%s/v%s", RoutePrefix(), spec.Info.Version), middlewares...), &h)
 
+	no_auth.RegisterHandlers(s.echo.Group(fmt.Sprintf("%s/v%s", RoutePrefix(), majorVersion), middlewaresNoAuth...), &h)
+	no_auth.RegisterHandlers(s.echo.Group(fmt.Sprintf("%s/v%s", RoutePrefix(), spec.Info.Version), middlewaresNoAuth...), &h)
```

Comment:
This looks scary, can't we use the same server, but have a new function, that will register only one handler?

Reply:
I didn't find an option to tell oapi-codegen to do so 🤔
and the function RegisterHandlers is generated

Reply:
hmm gotcha... and what about having this one endpoint added manually and only the rest would be generated?
I'm not sure if we need to even have the spec serving endpoint in the spec.

Reply:
manul = having extra file (for example `api_meta.go`) for the `RegisterSpecHandler` func

Reply:
I was more thinking that in the end we might add more endpoints to "noAuth" e.g. "/version" ?
(so avoid doing this manually)

And also having this information visible in the openAPI.yaml would be nice. That's why I wanted to use the security contexts to have it more visible but that didn't work out with the generator.

Also manually adding the route won't work as either all functions, handlers... are not generated (and need to be written manually) or it's already included in "RegisterHandlers" from what I saw oapi-codegen isn't flexible enough or I didn't find the switch to do it

Reply:
Does it make sense to show what I mean (changing the middleware functions) by creating a second DRAFT PR so we can choose?

Reply:
I think it can be very simple for this,  so we wouldn't need to rewrite from scratch IMO.
But yeah, implementing the handler that serves openapi spec manually, which should be pretty straight forward.
It feels a bit cleaner than having a separate generator just for this.

It's just a suggestion of course ;)

Reply:
Handling the security in the auth middleware itself sounds like a good option tho.